pkg/ebpf/initialization/kconfig.go

```diff
@@ -0,0 +1,60 @@
+package initialization
+
+import (
+	"fmt"
+	"github.com/aquasecurity/libbpfgo/helpers"
+	"os"
+)
+
+// Custom KernelConfigOption's to extend kernel_config helper support
+// Add here all kconfig variables used within tracee.bpf.c
+const (
+	CONFIG_ARCH_HAS_SYSCALL_WRAPPER helpers.KernelConfigOption = iota + helpers.CUSTOM_OPTION_START
+	CONFIG_FLATMEM
+	CONFIG_DISCONTMEM
+	CONFIG_SPARSEMEM
+	CONFIG_SPARSEMEM_VMEMMAP
+	CONFIG_SPARSEMEM_EXTREME
+	CONFIG_DYNAMIC_MEMORY_LAYOUT
+	CONFIG_X86_32
+	CONFIG_X86_PAE
+)
+
+var kconfigUsed = map[helpers.KernelConfigOption]string{
+	CONFIG_ARCH_HAS_SYSCALL_WRAPPER: "CONFIG_ARCH_HAS_SYSCALL_WRAPPER",
+	CONFIG_FLATMEM:                  "CONFIG_FLATMEM",
+	CONFIG_DISCONTMEM:               "CONFIG_DISCONTMEM",
+	CONFIG_SPARSEMEM:                "CONFIG_SPARSEMEM",
+	CONFIG_SPARSEMEM_VMEMMAP:        "CONFIG_SPARSEMEM_VMEMMAP",
+	CONFIG_SPARSEMEM_EXTREME:        "CONFIG_SPARSEMEM_EXTREME",
+	CONFIG_DYNAMIC_MEMORY_LAYOUT:    "CONFIG_DYNAMIC_MEMORY_LAYOUT",
+	CONFIG_X86_32:                   "CONFIG_X86_32",
+	CONFIG_X86_PAE:                  "CONFIG_X86_PAE",
+}
+
+// LoadKconfigValues load all kconfig variables used within tracee.bpf.c
+func LoadKconfigValues(kc *helpers.KernelConfig, isDebug bool) (map[helpers.KernelConfigOption]helpers.KernelConfigOptionValue, error) {
+	values := make(map[helpers.KernelConfigOption]helpers.KernelConfigOptionValue)
+	var err error
+	for key, keyString := range kconfigUsed {
+		if err = kc.AddCustomKernelConfig(key, keyString); err != nil {
+			return nil, err
+		}
+	}
+
+	// re-load kconfig and get just added kconfig option values
+	if err = kc.LoadKernelConfig(); err != nil { // invalid kconfig file: assume values then
+		if isDebug {
+			fmt.Fprintf(os.Stderr, "KConfig: warning: assuming kconfig values, might have unexpected behavior\n")
+		}
+		for key, _ := range kconfigUsed {
+			values[key] = helpers.UNDEFINED
+		}
+		values[CONFIG_ARCH_HAS_SYSCALL_WRAPPER] = helpers.BUILTIN // assume CONFIG_ARCH_HAS_SYSCALL_WRAPPER is a BUILTIN option
```

Comment:
Would you like to assume other values if KCONFIG is not present ? This is the case for alpine containers IIRC (not /proc/kconfig nor /boot/config-$(uname -r) files are available, and we're assuming the most common used values, like CONFIG_ARCH_HAS_SYSCALL_WRAPPER).